proposals/NNNN-noncopyable-generics.md

```diff
@@ -0,0 +1,713 @@
+```swift
+extension EventLog where Self: ~Copyable {
+  ...
+}
+```
+
+#### Protocol Inheritance
+
+A type must restate `~Copyable` even if it only conforms to protocols that are `~Copyable`, because the _absence_ of a requirement is not propagated:
+
+```swift
+protocol Token: ~Copyable {}
+
+// signature <Self where Self : Token, Self : Copyable>
+protocol ArcadeCoin: Token /* , Copyable */ {}
+
+// signature <Self where Self : Token>
+protocol CasinoChip: Token, ~Copyable {}
+```
+
+Note that `~Copyable` on a protocol is not viral:
+types that conform to a `~Copyable` protocol can themselves be `Copyable`.
+Similarly, a protocol that inherits from a noncopyable one can still require its
+conformers to be `Copyable`.
+
+In contrast, associated type requirements that are inherited from another protocol do preserve `~Copyable`.
+In the following example, `JobQueue.Job` remains noncopyable in `FIFOJobQueue`:
+```swift
+// signature <Self: Copyable>
+protocol JobQueue /* : Copyable */ {
+  associatedtype Job: ~Copyable
+  // ...
+}
+
+// signature <Self where Self: Copyable, Self: JobQueue>
+protocol FIFOJobQueue<Job>: JobQueue {
+  // associatedtype Job: ~Copyable // Because inherited
+  func pushBack(_ j: Job) // error: missing ownership specifier for parameter of
+                          //        noncopyable type 'Job'
+}
+```
+
+However, if the associated type requirement is  redeclared in the inheritor,
+the usual rule for `associatedtype` requirements will add an implicit `Copyable` requirement for the redeclaration:
+
+```swift
+// signature <Self where Self: JobQueue, Self.Job: Copyable>
+protocol LIFOJobQueue<Event>: JobQueue {
+  associatedtype Job /* : Copyable */
+}
+```
+
+## Detailed Design
+
+This section spells out additional details about the proposed extensions.
+
+### Conditionally-copyable types
+
+A generic type with properties that may or may not be copyable will often want
+to conditionally declare copyability.
+
+Since `Copyable` is an additive capability, this must be done by declaring
+the base type to not require copyability and then conditionally extending it:
+```swift
+struct MaybeCopyable<T: ~Copyable>: ~Copyable {
+  var t: T
+}
+
+extension MaybeCopyable /* : Copyable where T: Copyable */ { }
+```
+
+Conditional conformances to `Copyable` can only depend on
+generic parameters being `Copyable`.
+For example:
+```swift
+protocol P { }
+struct Foo<T: ~Copyable> { }
+
+extension Foo: P where T == Int { } // OK
+extension Foo: P where T: Copyable { } // OK
+extension Foo: Copyable where T: Copyable { } // OK
+extension Foo: Copyable where T: P { } // 🛑
+extension Foo: Copyable where T == Int {} // 🛑
+extension Foo: Copyable where T: Sendable {} // 🛑
+extension Foo: Copyable where T.A: Copyable {} // 🛑
+```
+
+### The top type
+
+The type `Any` is no longer the "top" type in the language, which is the type that is a supertype of all types.
```

Comment:
I'm not sure what the purpose of this section is. Can you remove it and instead show some of the valid and invalid subtype conversions for existentials in the following section? I think including this in the Existentials section makes more sense, because right now you're talking about existentials suppressing Copyable before introducing that as a concept.